### ISSUE TITLE
RuntimeError: maximum recursion depth exceeded

### ISSUE THREAD

**Issue by pikhovkin** [django-cache-machine/django-cache-machine#124]
I ran the test and got error.

Django==1.9.7

```
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 191, in fetch_by_id
    pks = [val[0] for val in vals]
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 258, in __iter__
    self._fetch_all()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1074, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 113, in __iter__
    obj = next(iterator)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 190, in fetch_by_id
    vals = self.values_list('pk', *list(self.query.extra.keys()))
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 725, in values_list
    clone = self._values(*fields)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 671, in _values
    clone = self._clone()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/caching/base.py", line 243, in _clone
    qs = super(CachingQuerySet, self)._clone(*args, **kw)
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/query.py", line 1059, in _clone
    query = self.query.clone()
  File "/home/user/.envs/cm_test/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 310, in clone
    obj._extra = self._extra.copy() if self._extra is not None else None
  File "/usr/lib/python2.7/collections.py", line 206, in copy
    return self.__class__(self)
  File "/usr/lib/python2.7/collections.py", line 69, in __init__
    self.__update(*args, **kwds)
  File "/home/user/.envs/cm_test/lib/python2.7/abc.py", line 151, in __subclasscheck__
    if subclass in cls._abc_cache:
  File "/home/user/.envs/cm_test/lib/python2.7/_weakrefset.py", line 72, in __contains__
    wr = ref(item)
RuntimeError: maximum recursion depth exceeded
```

**Comment by pikhovkin** [comment]
Error occurs when using the redis_byid settings.

**Comment by vkurup** [comment]
Thank you for the report. This should be (finally) addressed in the 1.0.0 release.
